src/account/operations/transfer/high_level/burn_foundry.rs

```diff
@@ -0,0 +1,203 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashSet;
+
+use iota_client::bee_message::{
+    output::{AliasId, AliasOutputBuilder, FoundryId, Output, OUTPUT_COUNT_MAX},
+    payload::transaction::TransactionId,
+};
+
+use crate::{
+    account::{
+        handle::AccountHandle, operations::transfer::TransferResult, types::OutputData, SyncOptions, TransferOptions,
+    },
+    Error,
+};
+
+impl AccountHandle {
+    /// Function to destroy foundry
+    pub async fn burn_foundry(
+        &self,
+        foundry_id: FoundryId,
+        options: Option<TransferOptions>,
+    ) -> crate::Result<TransferResult> {
+        log::debug!("[TRANSFER] burn_foundry");
+
+        let alias_id = *foundry_id.alias_address().alias_id();
+        let (existing_alias_output_data, existing_foundry_output_data) =
```

Comment:
Does this work every time without this check?
````
When Next State is empty, condition 1 and 2 are ignored. It must hold true, that Current State(Minted Tokens) + Token Diff = Current State(Melted Tokens)
````
from https://github.com/lzpap/tips/blob/master/tips/TIP-0018/tip-0018.md#simple-token-scheme-validation-rules before we can destroy it

Reply:
Just found out it doesn't always work, only worked when all native tokens had been melted, tried burning some and melting some and that didn't work as well.

So does this mean that a foundry must melt all of its minted tokens before it can be destroyed?

Reply:
I think so

Reply:
Hmm... But does this not make it almost impossible to destroy foundries? Because in most practical situations native tokens will be in circulation out of the foundries that minted them.

Reply:
I think that's true, if it wouldn't be like that, then there could be native tokens in circulation where one couldn't find the foundry for them to get the metadata

Reply:
Right, so this means that foundries that have some of their tokens burnt rather than melted can't ever be destroyed.

Reply:
Yes in this case we should just return an error I would say

Reply:
Is there a way to detect this for a token that was burnt without the knowledge of the foundry?

Reply:
No, but the protocol also doesn't allow to burn the foundry then, so if a user burns a native without the foundry, the foundry can never be destroyed

Reply:
And as such the controlling alias can never be destroyed as well.

Reply:
The alias could still be destroyed, but of course any control about the foundry would be lost then

Reply:
But we don't want to allow that do we? The foundry can't be unlocked without its controlling alias, could a foundry still be useful for keeping metadata in this case?

Reply:
Yes I would say better not allow it for not at least, if really wanted one could destroy it with a client lib or we might add something in the future